spring-pulsar/src/main/java/org/springframework/pulsar/core/TypedMessageBuilderCustomizer.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core;
+
+import java.util.function.Function;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+
+/**
+ * The function to customize a {@link TypedMessageBuilder}.
```

Comment:
I used an actual function in the backing issue, but that was mostly for illustration. I think we can use something similar to what Boot uses such as [here](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/DefaultKafkaProducerFactoryCustomizer.java). 

The 2 important things in there are 
* `@FunctionalInterface` marker on the interface
* Signature change slightly to `void customize(TheThingWeAreCustomizingHere<?, ?> theThing)`